de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/PowerSummaryTest.java

```diff
@@ -0,0 +1,102 @@
+import de.dlr.sc.virsat.model.extension.budget.power.model.PowerState;
+import de.dlr.sc.virsat.model.extension.budget.power.model.PowerSummary;
+import de.dlr.sc.virsat.model.extension.ps.model.ConfigurationTree;
+import de.dlr.sc.virsat.model.extension.ps.model.ElementConfiguration;
+
+public class PowerSummaryTest extends ASwtBotTestCase {
+	private SWTBotTreeItem repositoryNavigatorItem;
+	
+	private SWTBotTreeItem configurationTree;
+	private SWTBotTreeItem elementConfiguration1;
+	private SWTBotTreeItem elementConfiguration2;
+
+	private static final Double AOCS_POWER = 20.0;
+	private static final Double AOCS_DUTYCYCLE = 20.0;
+	
+	private static final Double DATA_HANDLING_POWER = 50.0;
+	private static final Double DATA_HANDLING_DUTYCYCLE = 50.0;
+
+	private SWTBotTreeItem powerEquipment1;
+	private SWTBotTreeItem powerEquipment2;
+	private SWTBotTreeItem powerSummary;
+
+	private Concept conceptPower;
+
+	
+	@Before
+	public void before() throws Exception {
+		super.before();
+		//create necessary items for the test
+		conceptPower = ConceptXmiLoader.loadConceptFromPlugin(de.dlr.sc.virsat.model.extension.budget.power.Activator.getPluginId() + "/concept/concept.xmi");
+		
+		repositoryNavigatorItem = bot.tree().expandNode(SWTBOT_TEST_PROJECTNAME, "Repository");
+
+		configurationTree = addElement(ConfigurationTree.class, conceptPs, repositoryNavigatorItem);
+
+		elementConfiguration1 = addElement(ElementConfiguration.class, conceptPs, configurationTree);
+		rename(elementConfiguration1, "AOCS");
+		
+		powerEquipment1 = addElement(PowerEquipment.class, conceptPower, elementConfiguration1);
+		openEditor(powerEquipment1);
+
+		bot.button("Add PowerState").click();
+		bot.table().select("powerValues");
+		bot.button("Drill-Down").click();
+		
+		renameField(PowerState.PROPERTY_POWER, AOCS_POWER + "");
+		renameField(PowerState.PROPERTY_DUTYCYCLE, AOCS_DUTYCYCLE + "");
+
+		elementConfiguration2 = addElement(ElementConfiguration.class, conceptPs, configurationTree);
+		rename(elementConfiguration2, "DataHandling");
+		
+		powerEquipment2 = addElement(PowerEquipment.class, conceptPower, elementConfiguration2);
+		openEditor(powerEquipment2);
+
+		bot.button("Add PowerState").click();
+		bot.table().select("powerValues");
+		bot.button("Drill-Down").click();
+		
+		renameField(PowerState.PROPERTY_POWER, DATA_HANDLING_POWER + "");
+		renameField(PowerState.PROPERTY_DUTYCYCLE, DATA_HANDLING_DUTYCYCLE + "");
+		save();
+		
+		powerSummary = addElement(PowerSummary.class, conceptPower, configurationTree);
+		openEditor(powerSummary);
+		save();
+	}
+	
+	@Test
+	public void calculationsForPowerSummaryTest() {
+		openEditor(configurationTree);
+		SWTBotTable allPropertyTable = getSWTBotTable(configurationTree, "Section for: PowerSummary");
+		String minPower = allPropertyTable.getTableItem(0).getText(2);
+		final String expectedMinPower = AOCS_POWER + DATA_HANDLING_POWER + "00 [W]";
+		assertEquals(expectedMinPower, minPower);
```

Comment:
If we open the powerSummary I think we can use the SWTBot API to do the assertion with something like this:
`assertText(expectedMinPower, bot.textWithLabel(PowerSummary.PROPERTY_MINPOWER));`
(or something similar)

In general it looks a bit weird that the before part opens the PowerSummary editor and then this test case opens the configuration tree editor.

Reply:
Ah ok, I thought that I was supposed to test, if the power summary in the configuration tree editor is showing the correct numbers.